src/cmd/support_archive/logger.go

```diff
@@ -0,0 +1,42 @@
+package support_archive
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	ctrlzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+const (
+	supportArchiveLoggerName = "[support-archive]"
+)
+
+func registerSupportArchiveLogger(ctx *supportArchiveContext) {
+	if ctx.toStdout {
+		// using stderr because we use stdout to deliver the tarball
+		ctx.log = newSupportArchiveLoggerToWriter(supportArchiveLoggerName, os.Stderr)
+	} else {
+		ctx.log = newSupportArchiveLoggerToWriter(supportArchiveLoggerName, os.Stdout)
+	}
+}
+
+func newSupportArchiveLoggerToWriter(name string, out io.Writer) logr.Logger {
```

Comment:
isn't the `name` param redundant as always the same value is passed into it?  (except in the tests)

Reply:
Exeactly, it's for testability.

Reply:
but what are you testing ?
that you can pass in a string and use it in another constructor ? (a string which is always the same constant in real use)

this code is just a copy-paste from the troubleshooting package, where this param makes sense as there are different "things" we test in the troubleshoot script and the logs should reflect it.(crd, secret, dynakube, ect.) 

but here its always the same

Reply:
Yeah, might be overengineered... I liked to change the logger name in the test, but I guess it's not strictly necessary.